File: DemoApplication.Web/src/app/models/auth/maker-checker.model.ts
// import { FileModel } from "./../file/file.model";
import { MakerCheckerActionsEnum } from "./../enums.model";

export class MakerCheckerModel {
    constructor(
        public id?: string, // actually guid
        public action?: MakerCheckerActionsEnum,
        public makerUser?: string, // actually guid
        public makerUserString?: string,
        public checkerUser?: string, // actually guid
        public checkerUserString?: string,
        public accepted?: boolean,
        public makerDate?: any,
        public checkerDate?: any,
        public motivation?: string,
        public files?: string[],
        public fileNames?: string[],
        public model?: string,
        public context?: string
    ) {
    }
}

File: DemoApplication.Models/MakerChecker/MakerCheckerModel.cs
using System;
using System.Collections.Generic;
using DemoApplication.Enums;

namespace DemoApplication.Models;

public class MakerCheckerModel
{
    public Guid Id { get; set; }
    public MakerCheckerActionsEnum Action { get; set; }
    public Guid MakerUser { get; set; }
    public Guid CheckerUser { get; set; }
    public bool Accepted { get; set; }
    public DateTimeOffset MakerDate { get; set; }
    public DateTimeOffset? CheckerDate { get; set; }
    public string Motivation { get; set; } = string.Empty;
    public IEnumerable<Guid> Files { get; set; } = Array.Empty<Guid>();
    public string Model { get; set; } = string.Empty;
}
